test/base.hpp

```diff
@@ -30,6 +30,12 @@
 #ifndef BASE_HPP_
 #define BASE_HPP_
 
+#if defined(__ARM_ARCH) && __ARM_ARCH <= 7
```

Comment:
I didn't develop this test, but it sort of seems like 1000 threads was arbitrarily chosen in the initial commit that added this test: ca6a6127835e06d7b75f353eb053d89f93dccb8f.  Personally, I'd just unconditionally reduce this to 500 threads, with a comment about why that particular number was chosen.  @nuclearsandwich , thoughts?